### PR TITLE
architecture: add simplified high-level diagram for flight stack

### PR DESCRIPTION
**Pull request by bkueng**
Similar to the one we had before but now with draw.io instead of mermaid

### CODE REVIEW

File: en/concept/architecture.md [hunk 1]
```diff
@@ -12,9 +12,24 @@ All PX4 [airframes](../airframes/README.md) share a single codebase (this includ
...
-## High Level Software Architecture
+## High-Level Software Architecture
 
-The diagram below provides an overview of the building blocks of PX4. 
+The following diagram shows a high-level overview for the building blocks of
+the flight stack. It contains the full pipeline from sensors, RC input and
+autonomous flight control (Navigator), down to the motor or servo control
+(Actuators).
+
+![PX4 High-Level Architecture](../../assets/diagrams/PX4_High-Level_Architecture.svg)
```
Comment (hamishwillee) [inline en/concept/architecture.md]: Does this render OK in gitbook? ON github it has some oddities: 
image

**Comment by hamishwillee** [comment]
Love this. Suggest restructure to have the original module diagram in its own section with a good title.
What I envisage is:
1. Where you currently have the text  "The diagram below provides a more detailed overview of the building blocks of PX4. ....."
  - Instead have a link to section down below with the image - e.g. Below LINK see a more detailed module level diagram.
  - Bring up JUST the sections on middleware and flight controller.
Essentially you're defining the architecture, then talking about the modules and benefits provided.

File: en/concept/architecture.md (hunk 1 above)
Comment (bkueng) [reply en/concept/architecture.md]: Yes. Interestingly it renders fine on github too with the direct link: https://github.com/PX4/Devguide/blob/3ab7ead214dded46ddeaa3f02cbcda27fce3cbc4/assets/diagrams/PX4_High-Level_Architecture.svg

**Comment by bkueng** [comment]
I rearranged the sections, feel free to add commits. I'm not quite sure if it's what you have envisaged. And I think 'High-Level Software Architecture' could now use a better title, do you have a good idea?

File: en/concept/architecture.md [hunk 2]
```diff
@@ -23,50 +21,57 @@ section shows the components of the flight stack.
...
+<!-- This diagram can be updated from 
+[here](https://drive.google.com/a/px4.io/file/d/15J0eCL77fHbItA249epT3i2iOx4VwJGI/view?usp=sharing) 
+and opened with draw.io Diagrams. You might need to request access if you
+don't have a px4.io Google account.
+Caution: it can happen that after exporting some of the arrows are wrong. In
+that case zoom into the graph until the arrows are correct, and then export
+again. -->
+
 An **estimator** takes one or more sensor inputs, combines them, and computes a
 vehicle state (for example the attitude from IMU sensor data).
```
Comment (hamishwillee) [inline en/concept/architecture.md:77]: Do you think it is worth having a paragraph here on the navigator role?

**Comment by hamishwillee** [comment]
Hi @bkueng,
I was thinking of the new diagram as a simplified version of the whole plaform, while it only covers the flight stack. Following my suggestion wasn't really possible on that basis, and IMO does not work as well as what you had.

What I have done is reverted the high level diagram and comments back to the top. I've kept the new diagram with the flight controller. I've made some other minor layout changes.

I think it is pretty good. Other than the question inline about navigator, I think it is good to merge :-)

File: en/concept/architecture.md (hunk 2 above)
Comment (bkueng) [reply en/concept/architecture.md:77]: Hmm, not sure. The others are more generic terms while the navigator is a specific module.

**Comment by bkueng** [comment]
Merged, thanks for your support!

**Comment by hamishwillee** [comment]
You're welcome. Nice one.

File: en/concept/architecture.md (hunk 2 above)
Comment (hamishwillee) [reply en/concept/architecture.md:77]: OK!

**Pull request merged**